superset-frontend/src/dashboard/components/filterscope/FilterScope.test.tsx

```diff
@@ -0,0 +1,372 @@
+    1: {
+      chartId: 1,
+      componentId: 'component-id',
+      datasourceId: 'datasource-id',
+      directPathToFilter: [],
+      isDateFilter: false,
+      isInstantFilter: false,
+      filterName: FILTER_A,
+      columns: { column_b: undefined, column_c: undefined },
+      labels: { column_b: FILTER_B, column_c: FILTER_C },
+      scopes: {
+        column_b: { immune: [], scope: [ROOT_ID] },
+        column_c: { immune: [], scope: [ROOT_ID] },
+      },
+    },
+  },
+  layout: {
+    ROOT_ID: { children: [GRID], id: ROOT_ID, type: 'ROOT' },
+    GRID: {
+      children: [TABS],
+      id: GRID,
+      type: GRID,
+      parents: [ROOT_ID],
+    },
+    TABS: {
+      children: [TAB_A, TAB_B],
+      id: TABS,
+      type: TABS,
+      parents: [ROOT_ID, GRID],
+    },
+    TAB_A: {
+      meta: { text: TAB_A },
+      children: [CHART_A, CHART_B],
+      id: TAB_A,
+      type: TAB,
+      parents: [ROOT_ID, GRID, TABS],
+    },
+    TAB_B: {
+      meta: { text: TAB_B },
+      children: [CHART_C, CHART_D],
+      id: TAB_B,
+      type: TAB,
+      parents: [ROOT_ID, GRID, TABS],
+    },
+    CHART_A: {
+      meta: {
+        chartId: 2,
+        sliceName: CHART_A,
+      },
+      children: [],
+      id: CHART_A,
+      type: 'CHART',
+      parents: [ROOT_ID, GRID, TABS, TAB_A],
+    },
+    CHART_B: {
+      meta: {
+        chartId: 3,
+        sliceName: CHART_B,
+      },
+      children: [],
+      id: CHART_B,
+      type: 'CHART',
+      parents: [ROOT_ID, GRID, TABS, TAB_A],
+    },
+    CHART_C: {
+      meta: {
+        chartId: 4,
+        sliceName: CHART_C,
+      },
+      children: [],
+      id: CHART_C,
+      type: 'CHART',
+      parents: [ROOT_ID, GRID, TABS, TAB_B],
+    },
+    CHART_D: {
+      meta: {
+        chartId: 5,
+        sliceName: CHART_D,
+      },
+      children: [],
+      id: CHART_D,
+      type: 'CHART',
+      parents: [ROOT_ID, GRID, TABS, TAB_B],
+    },
+  },
+  updateDashboardFiltersScope: jest.fn(),
+  setUnsavedChanges: jest.fn(),
+  onCloseModal: jest.fn(),
+});
+
+type CheckboxState = 'checked' | 'unchecked' | 'indeterminate';
+
+/**
+ * Unfortunatelly react-checkbox-tree doesn't provide an easy way to
+ * access the checkbox icon. We need this function to find the element.
+ */
+function getCheckboxIcon(element: HTMLElement): Element {
+  const parent = element.parentElement!;
+  if (parent.classList.contains('rct-text')) {
+    return parent.children[1];
+  }
+  return getCheckboxIcon(parent);
+}
+
+/**
+ * Unfortunatelly when using react-checkbox-tree, the only perceived change of a
+ * checkbox state change is the fill color of the SVG icon.
+ */
+function getCheckboxState(name: string): CheckboxState {
+  const element = screen.getByRole('link', { name });
+  const svgPath = getCheckboxIcon(element).children[1].children[0].children[0];
+  const fill = svgPath.getAttribute('fill');
+  return fill === '#20A7C9'
+    ? CHECKED
+    : fill === '#999999'
```

Comment:
If we use the `SupersetTheme` values here, this test won't break if/when the colors in the theme do.

Reply:
Great tip! Done.